pkg/webhook/core.oam.dev/v1alpha2/application/validation.go

```diff
@@ -31,8 +31,29 @@ import (
 	"github.com/oam-dev/kubevela/pkg/oam/util"
 )
 
-// ValidateCreate validates the Application on creation
-func (h *ValidatingHandler) ValidateCreate(ctx context.Context, app *v1beta1.Application) field.ErrorList {
+// ValidateWorkflow validates the Application workflow
+func (h *ValidatingHandler) ValidateWorkflow(ctx context.Context, app *v1beta1.Application) field.ErrorList {
+	var errs field.ErrorList
+	if app.Spec.Workflow != nil {
+		stepName := make(map[string]interface{})
+		for _, step := range app.Spec.Workflow.Steps {
+			if _, ok := stepName[step.Name]; ok {
+				errs = append(errs, field.Invalid(field.NewPath("spec", "workflow", "steps"), step.Name, "duplicate step name"))
+			}
+			stepName[step.Name] = nil
+			for _, sub := range step.SubSteps {
+				if _, ok := stepName[sub.Name]; ok {
+					errs = append(errs, field.Invalid(field.NewPath("spec", "workflow", "steps", "subSteps"), sub.Name, "duplicate step name"))
+				}
+				stepName[sub.Name] = nil
+			}
+		}
+	}
+	return errs
```

Comment:
Shall we parse the timeout format here? BTW, parse a string is not efficient, I think the duration can be a number with the unit `s` to avoid this check.